astropy/wcs/wcs.py

```diff
@@ -2449,22 +2448,83 @@ def sip_foc2pix(self, *args):
                    docstrings.RETURNS('pixel coordinates', 8))
 
     def proj_plane_pixel_scales(self):
-        """Same as :func:`astropy.wcs.utils.proj_plane_pixel_scales`
-        but returns a list of `~astropy.units.Quantity`.
-
         """
-        from astropy.wcs import utils as wcs_utils  # Avoid circular import
-        values = wcs_utils.proj_plane_pixel_scales(self)
+        Calculate pixel scales along each axis of the image pixel at
+        the ``CRPIX`` location once it is projected onto the
+        "plane of intermediate world coordinates" as defined in
+        `Greisen & Calabretta 2002, A&A, 395, 1061 <https://ui.adsabs.harvard.edu/abs/2002A%26A...395.1061G>`_.
+
+        .. note::
+            This method is concerned **only** about the transformation
+            "image plane"->"projection plane" and **not** about the
+            transformation "celestial sphere"->"projection plane"->"image plane".
+            Therefore, this function ignores distortions arising due to
+            non-linear nature of most projections.
+
+        .. note::
+            This method only returns sensible answers if the WCS contains
+            celestial axes, i.e., the `~astropy.wcs.WCS.celestial` WCS object.
+
+        Returns
+        -------
+        scale : list of `~astropy.units.Quantity`
+            A vector of projection plane increments corresponding to each
+            pixel side (axis).
+
+        See Also
+        --------
+        astropy.wcs.utils.proj_plane_pixel_scales
+
+        """  # noqa: E501
+        from astropy.wcs.utils import proj_plane_pixel_scales  # Avoid circular import
+        values = proj_plane_pixel_scales(self.celestial)
```

Comment:
Re-reading the docstring again, I think this is more correct? That is, by passing in `self.celestial` instead of `self`?

Reply:
There is nothing conceptually wrong with passing the "entire" `self`. Let's say `self` has 2 celestial axes and one spectral. This function would return something like `CDELT` for each axis. It should be fine. I think you should pass the whole WCS object.

Reply:
Ah, okay, I'll change it back then. This snippet from the `utils` docstring confused me.

```rst
    .. note::
        In order to compute the scales corresponding to celestial axes only,
        make sure that the input `~astropy.wcs.WCS` object contains
        celestial axes only, e.g., by passing in the
        `~astropy.wcs.WCS.celestial` WCS object.
```

Reply:
That is just to direct the reader what to do if that is what they want (i.e., to get the scale _only_ for celestial axes and exclude non-celestial axes)

Reply:
For the "area", it is highly unlikely that anyone would want to compute the area of 1 celestial axis and 1 spectral axis, even though mathematically one can do anything they want. Or, I am not sure most people would find it useful to compute the "volume" of a 2D celestial axes X 1D spectral. Probably, in most cases, user would want to separate these kind of axes but I may be wrong. If there is demand to compute volumes, let me know.

I personally find `proj_plane_pixel_scales` less useful than `proj_plane_pixel_area` because the later is used to approximate "mean pixel scale" (**geometric mean**) which, I personally find more useful than knowing scale along each axes. It also can be used to equalize pixel intensity when displaying an image (like it is done in FLT images).

Reply:
I did use `proj_plane_pixel_scales` when I was scratching my head over spacetelescope/stginga#205 😸